app/services/boarding_service.rb

```diff
@@ -0,0 +1,111 @@
+require "spaceship"
+
+class AddTesterResponse
+  attr_accessor :message
+  attr_accessor :type
+end  
+
+class BoardingService
+  include AbstractController::Translation
+
+  attr_accessor :app
+  attr_accessor :app_id
+  attr_accessor :user
+  attr_accessor :password
+  attr_accessor :tester_groups
+  attr_accessor :is_demo
+  attr_accessor :itc_token
+  attr_accessor :itc_closed_text
+
+  def initialize(app_id: ENV["ITC_APP_ID"],
+                   user: ENV["ITC_USER"] || ENV["FASTLANE_USER"],
+               password: ENV["ITC_PASSWORD"] || ENV["FASTLANE_PASSWORD"],
+          tester_groups: ENV["ITC_APP_TESTER_GROUPS"].to_s.split(/\s*,\s*/))
+    @app_id = app_id
+    @user = user
+    @password = password
+    @tester_groups = tester_groups
+    @is_demo = ENV["ITC_IS_DEMO"]
+    @itc_token = ENV["ITC_TOKEN"]
+    @itc_closed_text = ENV["ITC_CLOSED_TEXT"]
+
+    ensure_values
+  end
+
+  def add_tester(email, first_name, last_name)
+    tester = Spaceship::Tunes::Tester::External.find_by_app(app.apple_id, email)
+    add_tester_response = AddTesterResponse.new
+
+    if tester
+      Rails.logger.info "Tester #{email} already exists on application"
+      add_tester_response.message = t(:message_email_exists)
+      add_tester_response.type = "danger"
+    elsif tester = Spaceship::Tunes::Tester::External.find(email)
+      Rails.logger.info "Tester #{email} exists in iTunesConnect but not on application"
+      add_tester_response.message = t(:message_email_exists)
+      add_tester_response.type = "danger"
+    else
+      tester = Spaceship::Tunes::Tester::External.create!(email: email, first_name: first_name, last_name: last_name)
+      Rails.logger.info "Successfully created tester #{email}"
+      if testing_is_live?
+        add_tester_response.message = t(:message_success_live)
+      else
+        add_tester_response.message = t(:message_success_pending)
+      end
+      add_tester_response.type = "success"
+    end
+
+    if app.apple_id.length > 0
+      Rails.logger.info "Addding tester to application to groups: #{tester_groups.to_s}"
+      add_tester_to_groups!(tester: tester, app: app, groups: tester_groups)
+      Rails.logger.info "Done"
+    end
+    return add_tester_response
+  end
+
+  private
+    def ensure_values
+      raise "Environment variable `ITC_APP_ID` required" if @app_id.empty?
+      raise "Environment variable `ITC_USER` or `FASTLANE_USER` required" if @user.empty?
+      raise "Environment variable `ITC_PASSWORD` or `FASTLANE_PASSWORD`" if @password.empty?
```

Comment:
Yeah, I like the idea, great feedback @ohayon. You can use of more Ruby things, which would also add a space or new line inbetween the error messages:

```ruby
error_message = []
error_message << "Environment variable `ITC_APP_ID` required" if @app_id.empty?
error_message << "Environment variable `ITC_USER` or `FASTLANE_USER` required" if @app_id.empty?
...
raise error_message.join("\n")
```